drivers/Ticker.h

```diff
@@ -83,8 +83,11 @@ class Ticker : public TimerEvent, private NonCopyable<Ticker> {
      *  @param func pointer to the function to be called
      *  @param t the time between calls in seconds
      */
-    template <typename F>
-    MBED_FORCEINLINE void attach(F &&func, float t)
+#if defined(__ICCARM__)
```

Comment:
That's pretty grim, but it does appear to be the only forceinline template in the tree. I was thinking there were a bunch in the atomic stuff, but no.

I'd pull the `void attach(...)` out of the `#if` onto its own line.

Reply:
https://github.com/ARMmbed/mbed-os/compare/f65df2e951bf610254924c60cd354b6d75b18d7a..e2aeb601f14e922207350c65afb627a7ef2706f5